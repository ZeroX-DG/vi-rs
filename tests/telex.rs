mod shared;

fn snapshot_transform(lines: &str) -> String {
    shared::transform_lines(lines, |word| {
        let mut trasformed_word = String::new();
        vi::transform_buffer(&vi::TELEX, word.chars(), &mut trasformed_word);
        trasformed_word
    })
}

macro_rules! gen_test_telex {
    ($name:tt, $path: tt) => {
        gen_test!(snapshot_transform, $name, $path);
    };
}
gen_test_telex!(simple_telex, "../testdata/input/simple_telex.txt");
gen_test_telex!(
    non_vietnamese_telex,
    "../testdata/input/non_vietnamese_telex.txt"
);
