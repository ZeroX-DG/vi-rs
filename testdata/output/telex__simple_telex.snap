---
source: tests/telex.rs
expression: snapshot_transform(contents)
info:
  input_file: "../testdata/input/simple_telex.txt"
---
tôi là ai
đây là đâu
thêm một cái nữa
ví tiền
CHÀO
vơ
ru
vươn
chÊ
ă
zzzjjjjhhhkkk
chệch
chew
vươn
as
hoảng
được phát triển bởi cộng đồng lập trình viên hàng đầu VN
gì
được
ăn cơm chưa
Đây là
Đ$ay
xử lý
bây giờ
thế giới
hoặc
con cừu
dưới
dưới
hướng
daasd
aa
dd
chuw
chw
ựo
chuyện
quăng
tiếng việt
việt nam
xuất hiện xuất hiện xuất hiện
giàng ơi
Giàng ơi
ươ
ưo
ươn
hưo
huơ
thuở
được
trước
ngoao
ngoeo
ngoáo
ngoéo
giư
giữa
ha
trưo
trường
ươi
cười
hường
chửa
chuảwz
ừa
w
ư
Ư
